dropwizard-unix-socket/src/test/java/io/dropwizard/unixsocket/UnixSocketConnectorFactoryTest.java

```diff
@@ -0,0 +1,124 @@
+package io.dropwizard.unixsocket;
+
+import io.dropwizard.core.Application;
+import io.dropwizard.core.Configuration;
+import io.dropwizard.core.setup.Environment;
+import io.dropwizard.testing.ResourceHelpers;
+import io.dropwizard.testing.junit5.DropwizardAppExtension;
+import io.dropwizard.testing.junit5.DropwizardExtensionsSupport;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.StandardProtocolFamily;
+import java.net.UnixDomainSocketAddress;
+import java.nio.channels.Channels;
+import java.nio.channels.SocketChannel;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+@DisabledOnOs(OS.WINDOWS) // Windows has Unix Socket support but not on all versions.
+@ExtendWith(DropwizardExtensionsSupport.class)
+class UnixSocketConnectorFactoryTest {
+    private static final String httpRequest = "GET /app/hello HTTP/1.1\r\n" +
+        "Host: dropwizard-unixsock\r\n" +
+        "\r\n";
+
+    public DropwizardAppExtension<Configuration> ext = new DropwizardAppExtension<>(HelloWorldApplication.class,
+        ResourceHelpers.resourceFilePath("yaml/usock-server.yml"));
+
+    private UnixDomainSocketAddress socketAddress;
+
+    @BeforeEach
+    public void setUp() {
+        final File socket = new File("/tmp/dropwizard.sock");
+        socket.deleteOnExit();
+
+        socketAddress = UnixDomainSocketAddress.of(socket.getPath());
+    }
+
+    @Test
+    public void testClient() throws Exception {
+        try (SocketChannel channel = SocketChannel.open(StandardProtocolFamily.UNIX)) {
+            channel.connect(socketAddress);
+            OutputStream os = Channels.newOutputStream(channel);
+            OutputStreamWriter osw = new OutputStreamWriter(os, UTF_8);
+            PrintWriter writer = new PrintWriter(osw);
+            InputStream is = Channels.newInputStream(channel);
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, UTF_8))) {
+                writer.print(httpRequest);
+                writer.flush();
+
+                verifyHelloWorldReceived(reader);
+            }
+        }
+    }
+
+    @Test
+    public void testManyCalls() throws Exception {
```

Comment:
JUnit 5 test methods should be package-private

Reply:
![image](https://github.com/dropwizard/dropwizard/assets/64091305/00e59ea6-3d24-4249-8fb4-88b9aa4dc1c0)

Should I configure my Intellij for this?